ui.apps/src/main/content/jcr_root/apps/venia/clientlibs/common/js/CommerceGraphqlApi.js

```diff
@@ -0,0 +1,94 @@
+/*******************************************************************************
+ *
+ *     Copyright 2019 Adobe. All rights reserved.
+ *     This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License. You may obtain a copy
+ *     of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software distributed under
+ *     the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *     OF ANY KIND, either express or implied. See the License for the specific language
+ *     governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+
+(function () {
+
+    'use strict';
+    //todo update the proxy rules so that this gets router through the dispatcher
+    const imageUrlPrefix = "/magento/img";
+
+    class CommerceGraphqlApi {
+
+        constructor(props) {
+            if (!props.endpoint) {
+                throw new Error('The commerce API is not properly initialized. The "endpoint" property is missing from the initialization object');
+            }
+
+            this.endpoint = props.endpoint;
+        }
+
+        async _fetch(url, params) {
+            let response = await fetch(url, params);
+            if (!response.ok || response.errors) {
```

Comment:
I think we should not throw an error when `response.errors` exist. The GraphQL spec states that a response can contain `data` and `errors` at the same time, and we indeed see this with, for example, the Magento sample data where querying the category tree returns the data and some errors. So as long as there is some `data`, I think it's not fully an error.